test/files/neg/t10474.scala

```diff
@@ -0,0 +1,18 @@
+package t10474
+
+object Test {
+  def Foo(a: Int): Char = ???
+
+  object Bar
+
+  def crash[A](): Boolean = Bar match {
+    case Foo.Bar ⇒ true
+    case _ ⇒ false
+  }
+
+  type Crash = Foo.type
+
+  def bingo(l: Int, r: Int) = l + r
+
+  val `bingo`.Fuzz = 1
```

Comment:
Great job, I wanted to find some time to fix this issue. Currently, `def F(k:Int)=k;val F.K = ()` throws a `NullPointerException`.